core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -81,7 +89,38 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     new File(subDir, filename)
   }
 
-  def getFile(blockId: BlockId): File = getFile(blockId.name)
+  /** Looks up a file by blockManagerId */
+  // This method should be kept in sync with
+  // org.apache.spark.network.shuffle.StandaloneShuffleBlockManager#getFile().
+  def getFile(filename: String,  blockManagerId: BlockManagerId): File = {
```

Comment:
actually, I think it would make more sense to refactor this and the `getFile` in L65 to take the form:
```
def getLocalFile(
    filename: String,
    localDirs: Seq[String],
    createIfAbsent: Boolean = false): File
```
If the block manager is the one that belongs to this executor, then `createIfAbsent` will be true. Otherwise it's false.